### ISSUE TITLE
panicked at 'attempted to zero-initialize type `vorbisfile_sys::ov_callbacks`,

### ISSUE THREAD

**Issue by NofanTasi** [librespot-org/librespot#662]
Thanks for LibreSpot !
I upgraded CoreELEC from Leia to Matrix and LibreSpot stopped working

Here is the relevant log:

```
CoreGT:~/.kodi/temp # grep -i spot kodi.log
2021-03-04 13:27:39.651 T:5209     INFO <general>: CAddonMgr::FindAddons: service.librespot v19.0.126 installed
2021-03-04 13:35:47.661 T:6015     INFO <general>: Librespot: monitor started
2021-03-04 13:35:47.664 T:6015     INFO <general>: Librespot: ['librespot', '--backend', 'pulseaudio', '--bitrate', '320', '--cache', 'cache', '--device', 'librespot', '--device-type', 'TV', '--disable-audio-cache', '--name', 'Office', '--notify-kodi', '--autoplay']
2021-03-04 13:35:47.778 T:6015     INFO <general>: Librespot: loaded module-null-sink sink_name=librespot
2021-03-04 13:35:47.821 T:6015     INFO <general>: Librespot: loaded module-rtp-send destination_ip=127.0.0.1 port=24642 source=librespot.monitor
2021-03-04 13:35:47.857 T:6015     INFO <general>: Librespot: suspended sink 1
2021-03-04 13:35:47.858 T:6028     INFO <general>: Librespot: librespot thread started
2021-03-04 13:35:47.885 T:6028     INFO <general>: Librespot: librespot started
2021-03-04 13:35:47.913 T:6028     INFO <general>: Librespot: [2021-03-04T12:35:47Z INFO  librespot] librespot 81fbc79061 (2021-03-01). Built on 2021-03-01. Build ID: OEAk7aIq
2021-03-04 13:35:47.966 T:6028     INFO <general>: Librespot: *** WARNING *** The program 'librespot' uses the Apple Bonjour compatibility layer of Avahi.
2021-03-04 13:35:47.966 T:6028     INFO <general>: Librespot: *** WARNING *** Please fix your application to use the native API of Avahi!
2021-03-04 13:35:47.966 T:6028     INFO <general>: Librespot: *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
2021-03-04 13:35:48.049 T:6028     INFO <general>: Librespot: [2021-03-04T12:35:48Z INFO  librespot_core::session] Connecting to AP "gew1-accesspoint-e-n595.ap.spotify.com:4070"
2021-03-04 13:35:48.252 T:6028     INFO <general>: Librespot: [2021-03-04T12:35:48Z INFO  librespot_core::session] Authenticated as "xxxxxxxx" !
2021-03-04 13:35:48.255 T:6028     INFO <general>: Librespot: [2021-03-04T12:35:48Z INFO  librespot_core::session] Country: "BE"
2021-03-04 13:35:48.255 T:6028     INFO <general>: Librespot: country=BE
2021-03-04 13:36:42.841 T:6028     INFO <general>: Librespot: [2021-03-04T12:36:42Z INFO  librespot_connect::spirc] Fetching autoplay context uri
2021-03-04 13:36:42.864 T:6028     INFO <general>: Librespot: [2021-03-04T12:36:42Z INFO  librespot_connect::spirc] Autoplay uri resolved to <"spotify:station:artist:5pDjmC5mRl7vDJhsjVwNfk">
2021-03-04 13:36:42.888 T:6028     INFO <general>: Librespot: [2021-03-04T12:36:42Z INFO  librespot_playback::player] Loading <Ultraviolet - Live At St Pancras Old Church> with Spotify URI <spotify:track:5f3IwNA01M5jCGqqpe25ux>
2021-03-04 13:36:42.947 T:6028     INFO <general>: Librespot: thread '<unnamed>' panicked at 'attempted to zero-initialize type `vorbisfile_sys::ov_callbacks`, which is invalid', /rustc/cb75ad5db02783e8b0222fee363c5f63f7e2cf5b/library/core/src/mem/mod.rs:623:9
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot: suspended sink 1
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot: event panic 1/3
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot:    0: 0xab5c0908 - <unknown>
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot:    1: 0xab5eaeb0 - <unknown>
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot:    2: 0xab5b84cc - <unknown>
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot:    3: 0xab5c3130 - <unknown>
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot:    4: 0xab5c2b5c - <unknown>
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot:    5: 0xab5c3944 - <unknown>
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot:    6: 0xab5c346c - <unknown>
2021-03-04 13:36:42.993 T:6028     INFO <general>: Librespot:    7: 0xab5c0e84 - <unknown>
2021-03-04 13:36:42.994 T:6028     INFO <general>: Librespot:    8: 0xab5c33ec - <unknown>
2021-03-04 13:36:42.994 T:6028     INFO <general>: Librespot:    9: 0xab5e78e0 - <unknown>
2021-03-04 13:36:42.994 T:6028     INFO <general>: Librespot:   10: 0xab5e783c - <unknown>
2021-03-04 13:36:42.994 T:6028     INFO <general>: Librespot:   11: 0xab27d8b0 - <unknown>
2021-03-04 13:36:42.994 T:6028     INFO <general>: Librespot:   12: 0xab28ade4 - <unknown>
2021-03-04 13:36:42.994 T:6028     INFO <general>: Librespot:   13: 0xab288304 - <unknown>
2021-03-04 13:36:42.994 T:6028     INFO <general>: Librespot:   14: 0xab5c9f64 - <unknown>
2021-03-04 13:36:42.994 T:6028     INFO <general>: Librespot: [2021-03-04T12:36:42Z WARN  librespot_playback::player] Unable to load <SpotifyId { id: 228979858372245168803497515044361139625, audio_type: Track }>
2021-03-04 13:36:42.994 T:6028     INFO <general>: Librespot: Skipping to next track
2021-03-04 13:36:43.037 T:6028     INFO <general>: Librespot: suspended sink 1
2021-03-04 13:36:43.037 T:6028     INFO <general>: Librespot: librespot stopped
2021-03-04 13:36:43.055 T:6028     INFO <general>: Librespot: librespot started
2021-03-04 13:36:43.059 T:6028     INFO <general>: Librespot: [2021-03-04T12:36:43Z INFO  librespot] librespot 81fbc79061 (2021-03-01). Built on 2021-03-01. Build ID: OEAk7aIq
2021-03-04 13:36:43.086 T:6028     INFO <general>: Librespot: *** WARNING *** The program 'librespot' uses the Apple Bonjour compatibility layer of Avahi.
2021-03-04 13:36:43.087 T:6028     INFO <general>: Librespot: *** WARNING *** Please fix your application to use the native API of Avahi!
2021-03-04 13:36:43.087 T:6028     INFO <general>: Librespot: *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
2021-03-04 13:36:43.139 T:6028     INFO <general>: Librespot: [2021-03-04T12:36:43Z INFO  librespot_core::session] Connecting to AP "gew1-accesspoint-e-x4j5.ap.spotify.com:4070"
2021-03-04 13:36:43.317 T:6028     INFO <general>: Librespot: [2021-03-04T12:36:43Z ERROR librespot] Could not connect to server: bytes remaining on stream

```

**Comment by roderickvd** [comment]
This is not an issue of librespot, but of vorbis-rs upstream that triggers on newer Rust compilers: tomaka/vorbis-rs#19.

**Comment by NofanTasi** [comment]
thanks for pinpointing issue

**Comment by roderickvd** [comment]
Fix in PR tomaka/vorbis-rs#20.
